src/kudu/consensus/quorum_util.h

```diff
@@ -118,11 +118,15 @@ Status VerifyConsensusState(const ConsensusStatePB& cstate);
 // Provide a textual description of the difference between two consensus states,
 // suitable for logging.
 std::string DiffConsensusStates(const ConsensusStatePB& old_state,
-                                const ConsensusStatePB& new_state);
+                                const ConsensusStatePB& new_state,
+                                std::vector<std::string>* evicted_peers);
```

Comment:
these entire series of changes on DiffConsensusStates seems like an undesirable change.
If we can avoid this it would be really nice.

Reply:
The internal methods are not used anywhere else, so the surface area is small. The current config change flow does not expose/provide an easy way to fetch the peers that are removed from the ring easily - hence this approach. Also, evicted peers are not considered as state anywhere. On the commit of a config change, this method is called to detect the evicted peers (on the fly)